sms_sponsorship/forms/sms_sponsorship_registration_form.py

```diff
@@ -184,8 +184,10 @@ def form_after_create_or_update(self, values, extra_values):
             # validate sponsorship and send confirmation email
             sponsorship = self.main_object.sudo()
             sms_request = self.env['sms.child.request'].sudo().search([
-                ('sponsorship_id', '=', self.main_object.id)
+                ('partner_id', '=', sponsorship.partner_id.id)
```

Comment:
I'm not sure this is a good idea. Looking for the sponsorship id can only lead to the correct request while a partner may make several requests. I would keep sponsorship_id here.